service/src/main/java/io/camunda/service/license/CamundaLicense.java

```diff
@@ -7,41 +7,50 @@
  */
 package io.camunda.service.license;
 
-import org.apache.commons.lang3.StringUtils;
+import org.camunda.bpm.licensecheck.InvalidLicenseException;
+import org.camunda.bpm.licensecheck.LicenseKey;
+import org.camunda.bpm.licensecheck.LicenseKeyImpl;
 
 public class CamundaLicense {
 
-  public static final String LICENSE_ENV_VAR_KEY = "CAMUNDA_LICENSE_KEY";
-  public static String licenseStr;
-  private final EnvironmentVariableReader environmentVariableReader;
+  public String license;
+  public boolean isValid;
+  private volatile boolean isInitialized;
 
-  public CamundaLicense() {
-    environmentVariableReader = new EnvironmentVariableReader();
+  public CamundaLicense(final String license) {
+    this.license = license;
   }
 
-  /**
-   * Used for testing. Test can mock EnvironmentVariableReader to test out different environment
-   * variable values
-   */
-  public CamundaLicense(final EnvironmentVariableReader environmentVariableReader) {
-    this.environmentVariableReader = environmentVariableReader;
-  }
+  // for testing
+  public CamundaLicense() {}
 
   public boolean isValid() {
-    if (!isLicenseInitialized()) {
-      initializeLicenseCache();
-    }
+    initializeLicenseKey();
+    return isValid;
+  }
 
-    // TODO - return a real computed value, but always return true for now
-    return true;
+  protected void initializeLicenseKey() {
+    if (!isInitialized) {
+      synchronized (this) {
+        if (!isInitialized) {
+          isValid = determineLicenseValidity(license);
+          isInitialized = true;
+        }
+      }
+    }
   }
 
-  private boolean isLicenseInitialized() {
-    return StringUtils.isNotBlank(licenseStr);
+  protected boolean determineLicenseValidity(final String licenseStr) {
+    try {
+      final LicenseKey license = getLicenseKey(licenseStr);
+      license.validate(); // this method logs the license status
```

Comment:
🔧 Sorry but I feel this is really intransparent from the caller and a bit too deep for the specific concern that we want to log the license type. Furthermore, I feel like we are coupling two things here, which speaks against [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle). To me it seems we want to validate (not sure 100% what it means here tbh) and to log specific details about the license.

I think one should do the validation (or even give back what kind of license the key is either DEV or PROD, if not given default to DEV), and then outside we could log it.

Reply:
As for the dev/prod licenses, same response as above. There seems to be a discrepancy around dev/prod licenses

https://camunda.slack.com/archives/C06V2SR6U9G/p1722973721151959

Reply:
After looking into this some more, I'd argue that the current approach is what we want

One of the requirements for [the epic](https://github.com/camunda/product-hub/issues/2286) is: `As a Customer, my license information is logged as soon as C8 component (One Jar, Web Modeller or Console) starts.`

So instead of each component having to implement the logging capability on their own, the idea was to include this directly in the library - this has the advantage that the logging output is always consistent across applications.

Reply:
> this has the advantage that the logging output is always consistent across applications.

I see your point. I think this can also be argued in the other direction that depending on the component you might have different logging frameworks or details you want to add to the logging, so I would expect this is done outside of a lib. 

But anyway I don't think it is a blocker, just want to raise that we seem to couple here two features and that might be sub-optimal. I will leave the decision to you.